### PR TITLE
Refactor view sequencing and return sequenced views if recursion is found

### LINKED ISSUE #2494: [BUG]: Migrate Views With Dependencies
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Error trying to execute migrate-table task 

`19:32:58 ERROR [databricks.labs.ucx] {MainThread} Execute `databricks workspace export //Applications/ucx/logs/migrate-tables/run-392576937270338-0/migrate_views.log` locally to troubleshoot with more details. Invalid table references are preventing migration: {ViewToMigrate(src=Table(catalog='hive_metastore', database='xxx', name='yyy', object_type='VIEW', table_format='UNKNOWN', location=None `

In my env I have:
- more than 1k views.
- Some of them references at least one other view
- Some of them are created using mount points 

### Expected Behavior

view migrated because ucx codes handles some dependencies or view migration skipped .

Today the process abort instead of skipping and migrating the other views.

### Steps To Reproduce

1. ucx setup 0.33
2. Assessment run - databricks labs ucx ensure-assessment-run
3. Then Table mapping -  databricks labs ucx create-table-mapping
4. Create catalogs and schemas - databricks labs ucx create-catalogs-schemas
5. Then migrate-tables task - databricks labs ucx migrate-tables

### Cloud

Azure

### Operating System

macOS

### Version

latest via Databricks CLI

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by JCZuurmond**
## Changes
Refactor view sequencing and return sequenced views if recursion is found

### Linked issues
Resolves #2494

### Functionality

- [x] modified existing workflow: `table-migration`

### Tests

- [x] updated unit tests

### CODE REVIEW

**Review by JCZuurmond — COMMENTED** [review]
Did the refactor to improve code readability and to look for the possible origin of the problem

**Comment by JCZuurmond** [comment]
CI failure is unrelated. Fixed in https://github.com/databrickslabs/ucx/pull/2500

**Review by ericvergnaud — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/hive_metastore/view_migrate.py [hunk 1]
```diff
@@ -83,86 +83,104 @@ def __eq__(self, other):
...
+                continue
+            # If all dependencies are already processed, we can add the view to the next batch
+            not_processed_yet = set(view.dependencies) - views_from_previous_batches
+            if len(not_processed_yet) == 0:
+                result.append(view)
+                continue
+            if all(self._index.is_migrated(table_view.schema, table_view.name) for table_view in not_processed_yet):
+                result.append(view)
+        if len(result) == 0 and len(views) > 0:  # prevent infinite loop
+            raise RecursionError(f"Unresolved dependencies prevent batch sequencing: {views}")
```
Comment (nfx) [inline src/databricks/labs/ucx/hive_metastore/view_migrate.py:169]: ideally we'd not rely on exceptions for control flow, but let's continue with this one

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**